tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -1100,7 +1062,6 @@ void test_main(void)
 			 ztest_user_unit_test(test_user_workq_granted_access),
 			/* End order-important tests */
 			 ztest_1cpu_unit_test(test_work_submit_to_multipleq),
-			 ztest_unit_test(test_work_resubmit_to_queue),
```

Comment:
So now we don't support resubmitting from the callback, that I guaratee will break implementations.

Reply:
Yes, we do, whether from the work handler or externally.  That's part of the core API.  The only change to resubmission behavior is elimination of a potential livelock when resubmitting an already submitted item through the delayed work API would change its position in the work queue.

This (badly named) test was actually seeing whether there was an error return if the item was resubmitted through the delayed work API when it was being executed, i.e. in a case where cancellation had failed.  The way it forced the condition was to remove it from the queue so it appeared to be running.  Because work items can't be removed from the queue now there's no way to force the error, and in fact it's not even an error anymore.

The use of an error return from cancellation to indicate the work item was being processed was never robust: the item may well have completed before the caller gets the error return, and what's the caller supposed to do if it fails anyway?

Tracking "done" state in the handler callback itself, as demonstrated in the last commit, is the only reliable solution consistent with the documented API in the context of unknown priorities for the caller and the work queue threads.

Reply:
Yep, Im not agreeing with work items can't be removed from the queue now, that to me is a regression not a feature as you seem to be implying.